webextension/background.js

```diff
@@ -1,3 +1,174 @@
+const assignManager = {
+  closeableWindows: new Set([
+    "about:startpage",
+    "about:newtab",
+    "about:home",
+    "about:blank"
+  ]),
+  menuAssignId: "open-in-this-container",
+  menuRemoveId: "remove-open-in-this-container",
+  storageArea: browser.storage.local,
+
+  init() {
+    browser.tabs.onActivated.addListener((info) => {
+      browser.tabs.get(info.tabId).then((tab) => {
+        this.calculateContextMenu(tab);
+      }).catch((e) => {
+        throw e;
+      });
+    });
+
+    browser.windows.onFocusChanged.addListener((id) => {
+      browser.tabs.query({active: true}).then((tabs) => {
+        if (tabs && tabs[0]) {
+          this.calculateContextMenu(tabs[0]);
+        }
+      }).catch((e) => {
+        throw e;
+      });
+    });
+
+    browser.runtime.onMessage.addListener((request) => {
+      if (request.neverAsk === true) {
+        this.storageArea.set({neverAsk: true});
+      }
+    });
+
+    browser.contextMenus.onClicked.addListener((info, tab) => {
+      const pageUrl = new window.URL(info.pageUrl);
+      const userContextId = this.getUserContextIdFromCookieStore(tab);
+      // Mapping ${pageUrl.origin} to ${userContextId}
+      if (userContextId) {
+        const originStoreKey = this.getOriginStoreKey(pageUrl);
+        let actionName;
+        let storageAction;
+        if (info.menuItemId === this.menuAssignId) {
+          actionName = "added";
+          storageAction = this.storageArea.set({[originStoreKey]: userContextId});
+        } else {
+          actionName = "removed";
+          storageAction = this.storageArea.remove([originStoreKey]);
+        }
+        storageAction.then(() => {
+          browser.notifications.create({
+            type: "basic",
+            title: "Containers",
+            message: `Successfully ${actionName} ${pageUrl.origin} to always open in this container`,
```

Comment:
Ah, right ... http://gravatar.com and https://gravatar.com are treated differently. Alright then.